src/main/java/com/oracle/nosql/spring/data/core/NosqlTemplateBase.java

```diff
@@ -162,13 +167,27 @@ protected boolean doCreateTableIfNotExists(
         tableBuilder.append(")"); //create close )
 
         //ttl
-        if (entityInformation.getTtl() != null &&
-                entityInformation.getTtl().getValue() != 0) {
+        if (ttl != null && ttl.getValue() != 0) {
             tableBuilder.append(String.format(TEMPLATE_TTL_CREATE,
                     entityInformation.getTtl().toString()));
         }
         sql = tableBuilder.toString();
 
+        TableResult tableResult = doGetTable(entityInformation);
+        if (tableResult != null) {
+            /*table already exist in the database. Compare and throw error if
+            mismatch*/
+            String schema = tableResult.getSchema();
+            MapValue jsonSchema = JsonUtils.createValueFromJson(schema,
+                    new JsonOptions().setMaintainInsertionOrder(true)).asMap();
+            if (!compareTables(shardKeys, nonShardKeys, entityInformation,
+                    jsonSchema)) {
+                throw new IllegalArgumentException(String.format(
+                        "Error executing DDL '%s': Table %s exists" +
```

Comment:
This error is too general. I think the user needs to know a list of all mismatches that need fixing and hints on what to do to fix them.

Reply:
Changed to return specific error messages